app/src/main/java/com/example/nerdlauncher/ActivityAdapter.kt

```diff
@@ -0,0 +1,54 @@
+package com.example.nerdlauncher
+
+import android.content.Intent
+import android.content.pm.ResolveInfo
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+
+class ActivityAdapter(private val activities: List<ResolveInfo>) :
+    RecyclerView.Adapter<ActivityAdapter.ActivityHolder>() {
+
+    class ActivityHolder(itemView: View) :
+        RecyclerView.ViewHolder(itemView),
+        View.OnClickListener {
+        private val nameTextView = itemView as TextView
+        private lateinit var resolveInfo: ResolveInfo
+
+        init {
+            nameTextView.setOnClickListener(this)
+        }
+
+        fun bind(resolveInfo: ResolveInfo) {
+            this.resolveInfo = resolveInfo
+            val packageManager = itemView.context.packageManager
+            val appName = resolveInfo.loadLabel(packageManager).toString()
+            nameTextView.text = appName
+        }
+
+        override fun onClick(view: View?) {
```

Comment:
The logic for handling clicks should not be in the adapter if it is not UI logic.
E.g if when clicked, only the UI controlled by the adapter is changed, it's fine to do it here.
In cases such as here when we have business logic, we should handle this in the ViewModel or framgent

Reply:
You can do that by passing a lambda function in the constructor to handle clicks:
`    private val onItemClick: ((ItemModel) -> Unit)`